src/grandorgue/GrandOrgueFrame.cpp

```diff
@@ -476,6 +476,7 @@ void GOrgueFrame::Open(const GOrgueOrgan& organ)
 	GOrgueProgressDialog dlg;
 	m_doc = new GOrgueDocument(&m_Sound);
 	m_doc->Load(&dlg, organ);
+	UpdatePanelMenu();
 }
 
```

Comment:
1. When current organ is closed without opening the new one, the menu also should be updated (GOrgueFrame::DoClose() ?).
2. Because the panel menu is changing only when opening/closing an organ, calling UpdatePanelMenu() is no more necessary in GOrgueFrame::OnMenuOpen

Reply:
1. Perhaps, to be consistent - but it's not as problematic for an user as it is now either since it's empty anyway, large or small
2. No, it's still necessary to provide the ticks next to what panel(s) is/are open.

Reply:
1. Fixed
2. Won't fix, as this handles tick on/off for active or not active panels

Reply:
1. I see you have added it
2. Could you explain more?

    > to provide the ticks next to what panel(s) is/are open

    According to the code, UpdatePanelMenu fully destroys the old panel menu and constructs a new one and does not do anything more. This action is necessary when the list of pannels can be changed, but looks unnecessary on each opening the menu.

Reply:
When you have an organ opened the Panel menu is filled and each open panel will have a tick to the left of it to indicate that the panel is open. That tick is placed by the line
`item->Check(doc->WindowExists(GOrgueDocument::PANEL, panel) ? true : false);`
in the UpdatePaneMenu.
If the panel is not open there won't be a tick next to it. All that is taken care of in the UpdatePanelMenu function.

Reply:
Yes, you are right.

The best solution would be to split UpdatePaneMenu to two parts: the highweight constructing the menu - from Open and DoClose and the lighweigth item->Check for each panel - from GOrgueFrame::OnMenuOpen. But this refactoring is more than just bugfixing.

Reply:
That's my thoughts, exactly!